guides/v2.3/graphql/queries.md

```diff
@@ -1,35 +1,164 @@
+
+## Structure of a query
+
+A query contains the following elements:
+
+* The optional keyword `query`. If no keyword is specified at the beginning of a request, the processor assumes the request is a query.
+* An operation name for your local implementation. This name is required if you include variables. Otherwise, it is optional.
+* The query name
+* The terms to search for. The terms can be in the form of objects, attributes, or a combination. Queries that don't require search terms obtain their context from the customer's authorization token or store ID, both of which are specified in the header of the call. 
+* The output object, which specifies which data the query returns. 
+
+The following example shows the structure of the `cart` query:
+
+```text
+query myCartQuery{
+  cart(cart_id: String!): Cart
+}
+```
+
+In the preceding example, `myCartQuery` identifies your implementation of the `cart` query.  `cart_id` is a non-nullable string that defines the cart to query. (The exclamation point indicates the value is non-nullable.) The `Cart` output object defines which fields to return. 
+
+Now let's fully define a query:
+
+```text
+query myCartQuery{
+  cart(cart_id: "1WxKm8WUm3uFKXLlHXezew5WREfVRPAn") {
+    cart_id
+    items {
+      id
+      qty
+    }
+    billing_address {
+      firstname
+      lastname
+      postcode
+      }
+    shipping_addresses {
+      firstname
+      lastname
+      postcode
+    }
+  }
+}
+```
+
+In this example, we've supplied a cart ID as input, (which was generated by the `createEmptyCart` mutation). The output includes the `cart_id` as well as selected information about the items in the card and the billing and shipping addresses.
+
+The following example shows the query response:
+
+```json
+{
+  "data": {
+    "cart": {
+      "cart_id": "1WxKm8WUm3uFKXLlHXezew5WREfVRPAn",
+      "items": [
+        {
+          "id": "5",
+          "qty": 1
+        }
+      ],
+      "billing_address": {
+        "firstname": "Veronica",
+        "lastname": "Costello",
+        "postcode": "49628-7978"
+      },
+      "shipping_addresses": [
+        {
+          "firstname": "Veronica",
+          "lastname": "Costello",
+          "postcode": "49628-7978"
+        }
+      ]
+    }
+  }
+}
+```
+
+## Query variables 
+
+Specifying variables in a query can help increase code re-use. Consider the following requirements when generating a mutation that contains one or more variables:
```

Comment:
We're in a query topic discussing query variables, but here we're discussing a mutation. Can we change this to query?